bigquery/google/cloud/bigquery/table.py

```diff
@@ -473,8 +473,9 @@ def _build_resource(self):
             resource['schema'] = {
                 'fields': _build_schema_resource(self._schema)
             }
-        else:
-            raise ValueError("Set either 'view_query' or 'schema'.")
+        elif self.partitioning_type is None:
+            raise ValueError(
+                "Set either 'view_query' or 'schema' or 'partitioning_type'.")
```

Comment:
The original logic was not mine, so I didn't want to presume to change how it worked or what it checked (other than what was broken here that prevented create a partitioned table). Do you want me to remove these checks, and the tests that depend on them? (In my opinion, the ValueError does ensure the expected values are testable, so I'd argue against removing them completely.)

Re: "autodetect", I'm not sure that's true (though I'm not an expert here). This method creates an empty table (or view), so since there is no data from which to detect a schema, the schema is required. I believe (correct me if I'm wrong) that autodetect is used with a query that materializes the result to a job_destination, which is a different method.